quiz/quizzes/views.py

```diff
@@ -137,16 +137,25 @@ def list_player_quizzes(request: HttpRequest) -> HttpResponse:
     return render(request, 'user_quizzes.html', context)
 
 
-def choose_quiz_to_play(request: HttpRequest) -> HttpResponse:
+def choose_quiz_to_play(request):
     """Render the all quizzes created by all users page exclude this logged in"""
-    if request.user.is_authenticated:  # type: ignore
-        context = {
-            'quizzes': Quiz.objects.exclude(author=request.user)  # type: ignore
-        }
+    context = {'form': FilterQuizForm()}
+    quizzes = Quiz.objects.all() # type: ignore
+    if request.method == 'POST':
+        form = FilterQuizForm(request.POST)
+        if form.is_valid():
+            if form.cleaned_data['name']:
+                quizzes = quizzes.filter(name=form.cleaned_data['name'])
```

Comment:
Tu bym raczej zrobiła nie filtr z równa się, a jakiś filtr z zawieraniem (żaden użytkownik nie wpisze z palca dokładnego tytułu, za to wiele spróbuje wpisać jedno słowo i będzie oczekiwać że znajdzie coś niezależnie od np wielkości liter).